src/algebra/char_p/basic.lean

```diff
@@ -588,3 +588,16 @@ begin
 end
 
 end
+
+namespace ne_zero
+
+variables (R) (M : Type*) {r : R} {x y : M} {n p : ℕ} {a : ℕ+}
+
+lemma of_not_dvd [add_monoid_with_one M] [char_p M p] (h : ¬ p ∣ n) : ne_zero (n : M) :=
+⟨(not_iff_not.mpr $ char_p.cast_eq_zero_iff M p n).mpr h⟩
+
+lemma not_char_dvd [add_monoid_with_one R] (p : ℕ) [char_p R p] (k : ℕ) [h : ne_zero (k : R)] :
+  ¬ p ∣ k :=
```

Comment:
Do `M` and `R` have different typeclass assumptions? If not, why use two different variables?

Reply:
these were basically a copy + paste from the `ne_zero` file as it was too early in the imports now - I'll tidy this.